internal/db/identity.go

```diff
@@ -36,9 +37,13 @@ func (d *Database) GetIdentity(ctx context.Context, u *dbmodel.Identity) (err er
 	defer servermon.ErrorCounter(servermon.DBQueryErrorCount, &err, string(op))
 
 	db := d.DB.WithContext(ctx)
-	if err := db.Where("name = ?", u.Name).FirstOrCreate(&u).Error; err != nil {
+	result := db.Where("name = ?", u.Name).FirstOrCreate(&u)
+	if result.Error != nil {
 		return errors.E(op, err)
 	}
+	if result.RowsAffected == 1 {
```

Comment:
I'd put the logger on the *Database, then all can access it

Reply:
The function just calls `zapctx.Warn` or `zapctx.Error`, there's nothing to put on the db object.